### DISCUSSION TITLE
Conditionnal rendering with dynamic widgets

### DISCUSSION THREAD

**Discussion by Masjid74300Cluses** [algolia/instantsearch#5659] [Q&A]
I use dynamic widgets but I want to hide widgets (or panels) that, from the start, don't have any facets values.
Can it be done using the DynamicWidgets or the useDynamicWidgets hook ?

I want to achieve something like this : 
```
<DynamicWidgets>
  <RefinementList attribute="brand" />
  <RefinementList attribute="size" onlyIfFacetValues /> 
</DynamicWidgets>

**Comment by dhayab** [comment]
If you know in advance which facets apply to which type of results, you can create Rules to dynamically return facets that apply to known conditions: https://www.algolia.com/doc/guides/managing-results/rules/merchandising-and-promoting/how-to/merchandising-facets/#merchandise-facet-values-in-the-visual-editor.

Otherwise, you could rely on `useDynamicWidgets()` and [`useInstantSearch()`](https://www.algolia.com/doc/api-reference/widgets/use-instantsearch/react-hooks/#widget-param-results) in the same React component. The latter will give you access to the `results` so you'll be able to determine which facets are empty, and return the corresponding refinement UI accordingly.
